From 0378326e4dd78d640fa3b087765382a1109feb65 Mon Sep 17 00:00:00 2001
From: Gabriele M <moto.falcon.git@gmail.com>
Date: Tue, 6 Jun 2017 22:13:02 +0200
Subject: [PATCH 117/173] Snap: Simulate back button press when menu back
 button is pressed

Launching Gallery when the home button (shown as back button) is
pressed makes no sense, especially because it's only shown in Snap's
own gallery. In addition to that, if Gallery is disabled, Snap closes
itself. Fix both the issues by simply simulating a back button press
when the menu back button is pressed.

REGRESSION-607

Change-Id: I0b2b6ee074715e2866bcebb009118b8271b1efdf
---
 src/com/android/camera/CameraActivity.java | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/src/com/android/camera/CameraActivity.java b/src/com/android/camera/CameraActivity.java
index d1ee7754a..ec6d7306b 100755
--- a/src/com/android/camera/CameraActivity.java
+++ b/src/com/android/camera/CameraActivity.java
@@ -1379,14 +1379,8 @@ public class CameraActivity extends Activity
         // Handle presses on the action bar items
         switch (item.getItemId()) {
             case android.R.id.home:
-                // ActionBar's Up/Home button was clicked
-                try {
-                    startActivity(IntentHelper.getGalleryIntent(this));
-                    return true;
-                } catch (ActivityNotFoundException e) {
-                    Log.w(TAG, "Failed to launch gallery activity, closing");
-                    finish();
-                }
+                onBackPressed();
+                return true;
             case R.id.action_delete:
                 UsageStatistics.onEvent(UsageStatistics.COMPONENT_CAMERA,
                         UsageStatistics.ACTION_DELETE, null, 0,
-- 
2.17.1

